dlib/image_transforms/interpolation.h

```diff
@@ -1543,33 +1546,32 @@ namespace dlib
             << "\n\t objects2.size():   " << objects2.size() 
             );
 
-        image_array_type new_images;
-        std::vector<std::vector<T> > new_objects;
-        std::vector<std::vector<U> > new_objects2;
+        using Objects = std::vector<std::vector<T>>;
+        using Objects2 = std::vector<std::vector<U>>;
 
-        using namespace impl; 
+        using namespace impl;
 
-        std::vector<T> objtemp;
-        std::vector<U> objtemp2;
-        typename image_array_type::value_type temp;
-        for (long i = 0; i < angles.size(); ++i)
-        {
-            for (unsigned long j = 0; j < images.size(); ++j)
+        image_array_type new_images(images.size() * angles.size());
+        Objects new_objects(images.size() * angles.size());
```

Comment:
Don't make the Objects typedef. You only use it once.  So adding it is just more code and more confusing indirection for someone reading it.  Just say std::vector<std::vector<T>>.  That's much clearer.

Reply:
Agreed. Fixed.